ddev/ddev.com-front-end
# Add featured sponsors to Support DDEV page

## The Issue

#44 raised the issue of featured sponsors not appearing at all on the Support DDEV page’s “Sponsor Development” section.

## How This PR Solves The Issue

This adds those featured sponsors to the beginning of that list and uses GitHub handles to make sure they’re not repeated.

Before and after:

![Screen Shot 2023-04-21 at 06 29 12 PM@2x](https://user-images.githubusercontent.com/2488775/233755278-006c71d2-76f3-4065-88c3-6b95af1505b1.png)

> 💡 I thought it was a bug that Peter Kraume shows up twice, but he’s sponsoring both `rfay` and `ddev`.

It also does a few things that’ll make it easier to acknowledge sponsorship elsewhere:

- Moves featured sponsor data out of a component and into a standalone [`featured-sponsors.json`](https://github.com/ddev/ddev.com-front-end/blob/feature/support-featured-sponsors/src/featured-sponsors.json) file that can be pulled into different contexts.
- Includes a sponsor `"type"`, currently `"major"` or `"standard"`—which ideally we could map to consistent sponsorship tiers. With that done, visual treatment in different situations could be easier to manage.
- Adds square SVG logos for each sponsor where they were necessary, for more layout control. (Also easy to PR if I committed any brand guide offenses, though I tried to be careful.)

## Manual Testing Instructions

Try it locally or check out the [preview build](https://cbb82392.ddev-com-front-end.pages.dev/support-ddev/).

## Automated Testing Overview

n/a

## Related Issue Link(s)

- #44

## Release/Deployment Notes

n/a
